test/components/popup.spec.tsx

```diff
@@ -117,6 +117,31 @@ describe('<Popup />', () => {
         });
     });
 
+    it('syncs the popup position when rendered before the anchor and the anchor size changes', async () => {
+        class Dummy extends React.PureComponent {
+            public state = {anchor: null};
+
+            public render() {
+                return (
+                    <div>
+                        <Popup anchor={this.state.anchor} open syncWidth>
+                            <div data-automation-id="POPUP">Popup body</div>
+                        </Popup>
+                        <div data-automation-id="ANCHOR" ref={node => this.setState({anchor: node})}>
+                            {this.props.children}
+                        </div>
+                    </div>
+                );
+            }
+        }
+
+        const select = selectDom(document.body);
```

Comment:
Is there any way to use the popup driver in this test? Maybe create a simple dummy driver? I think we should try no using selectDom / body select and so on.

Reply:
If I'm not mistaken, the purpose of drivers is to decouple tests from the internal DOM structure of the components. I don't see a violation of this principle here. The test is aware of the structure of the component defined within the test itself. This is like a class instance being aware of its private properties.